src/arrayMethodJoin.js

```diff
@@ -5,7 +5,29 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const arrLength = this.length;
+    let result = '';
+    let newSeparator = separator;
+
+    if (separator === undefined) {
+      newSeparator = ',';
+    }
```

Comment:
I see that you are assignin separator to a new value (line 10) because you want to reasign it (line 13), but you can consider asigning a default value for the separator, like so:

= function(separator = ',') [line 7]